packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -325,69 +325,74 @@ const numberWithUnitCellView = NumberWithUnitCellView.compose({
 
 ```TS
-type TableColumnButtonCellData = TableStringField<'id'>;
-
-public class TableColumnButton extends TableColumn<TableColumnButtonCellData> {
+export interface ColumnInternalsOptions {
     ...
+    readonly delegatedEvents: readonly string[];
+    ...
+}
 
-    public cellRecordFieldNames = ['id'] as const;
-
-    @attr
-    public idKey: string;
-
-    public getDataRecordFieldNames(): string[] {
-        return [valueKey];
-    }
-
-    public cellViewTag = 'table-cell-view-button';
-
-    public validateCellData(cellData: TCellRecord): void {
-        if (typeof(cellData['id']) !== 'string') {
-            throw new Error('Type for cellData is incorrect!');
-        }
+export class ColumnInternals<TColumnConfig> {
+    ...
+    public readonly delegatedEvents: readonly string[];
+    ...
+    public constructor(options: ColumnInternalsOptions) {
+        this.delegatedEvents = options.delegatedEvents;
+        ...
     }
 }
 
-class ButtonCellView extends TableCellView<TableColumnButtonCellData> {
-    public onButtonClick(): void {
-        this.$emit('button-click', { data: this.cellRecord.id })
+AnchorTableColumn extends TableColumn {
+    constructor() {
+        super({
+            delegatedEvents: ['click'],
+            ...
+        });
     }
+    ...
 }
-const buttonCellView = ButtonCellView.compose({
-    baseName: 'table-cell-view-button',
-    template: html<ButtonCellView>`
-        <nimble-button @click="${(x) => x.onButtonClick()}">
-            <span>Press Me</span>
-        </nimble-button>`,
-    styles: /* styling */
-});
 ```
 
-Angular template:
+Upon connecting a cell view to the DOM, we will attach an event listener for each event type in `columnInternals.delegatedEvents`. The handler will wrap the original event in a new `CustomEvent` and dispatch that to the column.
 
-```HTML
-<nimble-table (button-click)="doSomething($event)">
-    <nimble-table-column-button></nimble-table-column-button>
-</nimble-table>
+```TS
+this.addEventListener(delegatedEventName, event => {
+    this.column.dispatchEvent(new CustomEvent('delegated-event', {
+        details: {
+            originalEvent: event
+        }
+    }));
+});
 ```
 
-Angular code:
+The `CustomEvent`'s details will be of type `DelegatedEventEventDetails`:
 
 ```TS
-private doSomething(CustomEvent e): void {
-    ...
+export interface DelegatedEventEventDetails {
+    originalEvent: Event;
 }
-
 ```
 
-The exact pattern for how we expect event APIs to be implemented is TBD. The above is simply illustrative of one approach, but it's safe to say that the goal will be to provide frameworks like Angular the expected event binding APIs.
--->
+A column type (or an Angular directive on that type) may register a listener for the `delegated-event` event. This listener will have access to the full original event, including the originating cell view via `event.target`. The cell view can expose anything necessary from its public API.
```

Comment:
The first part of this first sentence is a bit confusing to me.

> A column type ... may register a listener for the `delegated-event` event

A given column type is never registering an event listener for the delegated events, right? Should this really say something like _A client may register a listener for the `delegated-event` event on a table column_?